hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/MariaDBLegacySqlAstTranslator.java

```diff
@@ -157,6 +160,75 @@ else if ( expression instanceof Summarization ) {
 		}
 	}
 
+	@Override
+	public void visitLikePredicate(LikePredicate likePredicate) {
+		if ( likePredicate.isCaseSensitive() ) {
+			likePredicate.getMatchExpression().accept( this );
+			if ( likePredicate.isNegated() ) {
+				appendSql( " not" );
+			}
+			appendSql( " like " );
+			escapeLikePattern( likePredicate.getPattern(), likePredicate.getEscapeCharacter() );
+		}
+		else {
+			appendSql( getDialect().getLowercaseFunction() );
+			appendSql( OPEN_PARENTHESIS );
+			likePredicate.getMatchExpression().accept( this );
+			appendSql( CLOSE_PARENTHESIS );
+			if ( likePredicate.isNegated() ) {
+				appendSql( " not" );
+			}
+			appendSql( " like " );
+			appendSql( getDialect().getLowercaseFunction() );
+			appendSql( OPEN_PARENTHESIS );
+			escapeLikePattern( likePredicate.getPattern(), likePredicate.getEscapeCharacter() );
+			appendSql( CLOSE_PARENTHESIS );
+		}
+		if ( likePredicate.getEscapeCharacter() != null ) {
+			appendSql( " escape " );
+			likePredicate.getEscapeCharacter().accept( this );
+		}
+	}
+
+	protected void escapeLikePattern(Expression pattern, Expression escapeCharacter) {
```

Comment:
I think it would be good to move this method to the `AbstractSqlAstTranslator` to avoid duplication of the logic and name it `renderBackslashEscapedLikePattern` as well as pass a `noBackslashEscapesEnabled` parameter.

Reply:
Sounds good to me